packages/integration-sdk-runtime/src/execution/jobState.ts

```diff
@@ -191,24 +142,40 @@ export function createStepJobState({
   afterAddEntity,
   afterAddRelationship,
   uploader,
+  logger,
```

Comment:
Do we want to propagate this hook all the way up? In my updates I created the hook right before the jobState is created, but can  raise it further if we think it will be useful.